File: .claude/hooks/notification-sound.mjs
#!/usr/bin/env node
// @hookstack notification-sound
// Joue un son système quand Claude attend l'utilisateur (Notification)
// Sur macOS, clic sur la notif ramène le bon contexte (terminal ou Claude app)
import { execSync } from 'node:child_process';
import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';

// Bundle IDs des terminaux courants — clic ramène le bon contexte
const TERMINAL_BUNDLE_IDS = {
  'iTerm.app':       'com.googlecode.iterm2',
  'Apple_Terminal':  'com.apple.Terminal',
  'WezTerm':         'com.github.wez.wezterm',
  'ghostty':         'com.mitchellh.ghostty',
  'vscode':          'com.microsoft.VSCode',
  'cursor':          'com.todesktop.230313mzl4w4u92',
};
const CLAUDE_APP_BUNDLE_ID = 'com.anthropic.claudefordesktop';

export function resolveActivateBundle(termProgram) {
  return TERMINAL_BUNDLE_IDS[termProgram] ?? CLAUDE_APP_BUNDLE_ID;
}

function defaultExec(cmd) {
  execSync(cmd, { timeout: 5_000, stdio: 'ignore', shell: true });
}

function defaultHasTerminalNotifier() {
  try { execSync('which terminal-notifier', { timeout: 2_000, stdio: 'pipe', shell: true }); return true; }
  catch { return false; }
}

export function run(_input, {
  exec = defaultExec,
  hasTerminalNotifier = defaultHasTerminalNotifier,
  platform = process.platform,
  termProgram = process.env.TERM_PROGRAM,
} = {}) {
  try {
    if (platform === 'darwin') {
      const bundleId = resolveActivateBundle(termProgram);

      if (hasTerminalNotifier()) {
        // Clic sur la notif → ramène automatiquement le bon contexte
        exec(
          `terminal-notifier -title "Claude Code" -message "Claude needs your input" -activate ${bundleId} -sound Glass`
        );
      } else {
        // Fallback : son + notification sans action de clic (brew install terminal-notifier pour activer)
        exec('afplay /System/Library/Sounds/Glass.aiff');
        exec('osascript -e \'display notification "Claude needs your input" with title "Claude Code"\'');
      }
    } else if (platform === 'linux') {
      exec('paplay /usr/share/sounds/freedesktop/stereo/message.oga 2>/dev/null || aplay /usr/share/sounds/alsa/Front_Center.wav 2>/dev/null || true');
    } else if (platform === 'win32') {
      exec('powershell -c "[console]::beep(660, 300)"');
    }
  } catch {
    // Son absent ou erreur — non bloquant
  }
  return null;
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  run(input);
}

File: .claude/hooks/notification-sound.py
#!/usr/bin/env python3
# @hookstack notification-sound
"""Plays a system sound when Claude waits for the user (Notification).
On macOS, clicking the notification brings back the right context (terminal or Claude app).
"""
import json
import os
import subprocess
import sys

# Bundle IDs des terminaux courants — clic ramène le bon contexte
TERMINAL_BUNDLE_IDS = {
    "iTerm.app": "com.googlecode.iterm2",
    "Apple_Terminal": "com.apple.Terminal",
    "WezTerm": "com.github.wez.wezterm",
    "ghostty": "com.mitchellh.ghostty",
    "vscode": "com.microsoft.VSCode",
    "cursor": "com.todesktop.230313mzl4w4u92",
}
CLAUDE_APP_BUNDLE_ID = "com.anthropic.claudefordesktop"


def resolve_activate_bundle(term_program):
    return TERMINAL_BUNDLE_IDS.get(term_program, CLAUDE_APP_BUNDLE_ID)


def _default_exec(cmd):
    subprocess.run(
        cmd, shell=True, timeout=5, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
    )


def _default_has_terminal_notifier():
    try:
        subprocess.run(
            "which terminal-notifier",
            shell=True,
            timeout=2,
            check=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
        )
        return True
    except Exception:
        return False


def run(
    input_data=None,
    *,
    exec_cmd=None,
    has_terminal_notifier=None,
    platform=None,
    term_program=None,
):
    if exec_cmd is None:
        exec_cmd = _default_exec
    if has_terminal_notifier is None:
        has_terminal_notifier = _default_has_terminal_notifier
    if platform is None:
        platform = sys.platform
    if term_program is None:
        term_program = os.environ.get("TERM_PROGRAM")

    try:
        if platform == "darwin":
            bundle_id = resolve_activate_bundle(term_program)

            if has_terminal_notifier():
                # Clic sur la notif → ramène automatiquement le bon contexte
                exec_cmd(
                    f'terminal-notifier -title "Claude Code" -message "Claude needs your input" '
                    f"-activate {bundle_id} -sound Glass"
                )
            else:
                # Fallback : son + notification sans action de clic (brew install terminal-notifier pour activer)
                exec_cmd("afplay /System/Library/Sounds/Glass.aiff")
                exec_cmd(
                    'osascript -e \'display notification "Claude needs your input" with title "Claude Code"\''
                )
        elif platform == "linux":
            exec_cmd(
                "paplay /usr/share/sounds/freedesktop/stereo/message.oga 2>/dev/null || "
                "aplay /usr/share/sounds/alsa/Front_Center.wav 2>/dev/null || true"
            )
        elif platform == "win32":
            exec_cmd('powershell -c "[console]::beep(660, 300)"')
    except Exception:
        # Son absent ou erreur — non bloquant
        pass
    return None


if __name__ == "__main__":
    data = json.load(sys.stdin)
    run(data)
